metabrainz/model/payment.py

```diff
@@ -160,14 +166,29 @@ def process_paypal_ipn(cls, form):
             logging.info('PayPal: Payment not completed. Status: "%s".', form['payment_status'])
             return
 
+        account_ids = current_app.config['PAYPAL_ACCOUNT_IDS']  # "currency => account" mapping
+
+        if form['mc_currency'].lower() not in SUPPORTED_CURRENCIES:
+            logging.warning("Unsupported currency: ", form['mc_currency'])
+            return
+
+        # Checking if payment was sent to the right account depending on the currency
+        if form['mc_currency'].upper() in account_ids:
```

Comment:
When money is sent to an account for another currency. If that's too redundant, I can remove the check.